### DISCUSSION TITLE
Multiline description to generated schema

### DISCUSSION THREAD

**Discussion by sunghwan-daim** [ChilliCream/hotchocolate#5561] [Feature Request]
Using HotChocolate as a GraphQL backend, it would be great if it could generate a schema file with a block string description for better readability.

Given code
```cs
public class Query {
    [GraphQLDescription("""
        abc
        def
        """)]
    public string GetApplicationName() => "app";
}
```

generates a schema like:
```graphql
type Query {
  """
  abc
  def
  """
  applicationName: String!
}
```

**Comment by PascalSenn** [comment]
I am not sure if i understand this correctly. This works already as you describe.

**Reply by sunghwan2789** [reply]
I made a small [repo](https://github.com/sunghwan2789/gql-playground/tree/7dc61cd8ded8cc738fe01f348ec12ee7d26af11e).

When I download schema by visiting `http://127.0.0.1:5294/graphql?sdl`, HotChocolate outputs single-line descriptions with ugly `\n` character (`\r\n` in CRLF encoding btw).

```gql
schema {
  query: BookService
  subscription: Subscriptions
}

type Book {
  id: Int!
}

"hi\nhello"
type BookService {
  "hello\nworld"
  books(count: Int!): [Book!]!
}

type Subscriptions {
  bookPublished(seed: Int!): Book!
}
```

I want `"hi\nhello"` to be
```gql
"""
hi
hello
"""
```
Thanks!

**Comment by Nathan-Molby** [comment]
Did this get completed? It's related to an issue in an Apollo repo here: https://github.com/apollographql/apollo-ios/issues/3553#issuecomment-2860837258

**Reply by sunghwan2789** [reply]
I think there is a test for this, so it should work...:
https://github.com/ChilliCream/graphql-platform/blob/b7c00cffae91f5525140e08f6889e3385d13fcfa/src/HotChocolate/Core/test/Types.Tests/Types/DescriptionTests.cs#L306-L311

https://github.com/ChilliCream/graphql-platform/blob/b7c00cffae91f5525140e08f6889e3385d13fcfa/src/HotChocolate/Core/test/Types.Tests/Types/__snapshots__/DescriptionTests.Schema_With_All_Possible_Descriptions.graphql#L11-L15

**Reply by sunghwan2789** [reply]
How do you get the schema? by ast? by url?
By url, this is working: `http://localhost/graphql/schema.graphql`

**Reply by Nathan-Molby** [reply]
Sorry, I didn't see that this was marked as resolved, I just saw that it was closed. We are using an older version of HotChocolate and will need to update to 14.0 to fix this. Thank you.
